subprojects/snapshots/src/main/java/org/gradle/internal/vfs/impl/DefaultFileSystemAccess.java

```diff
@@ -93,23 +93,23 @@ public <T> Optional<T> readRegularFileContentHash(String location, Function<Hash
                 }
                 return Optional.empty();
             })
-            .orElseGet(() -> {
+            .orElseGet(() -> virtualFileSystem.store(location, vfsStorer -> {
                 File file = new File(location);
                 FileMetadata fileMetadata = this.stat.stat(file);
                 if (fileMetadata.getType() == FileType.Missing) {
-                    virtualFileSystem.store(location, () -> new MissingFileSnapshot(location, fileMetadata.getAccessType()));
+                    vfsStorer.store(new MissingFileSnapshot(location, fileMetadata.getAccessType()));
```

Comment:
Don't we know the hash to return here already?

Reply:
We run `null` here, since the file is a missing file. We store the missing file snapshot in the VFS for performance reasons, aka then we can re-use it.

Reply:
Ah, I see. The conditional flow is a bit weird, though. Can we switch to a `switch` maybe?

Reply:
Using switch now.